tests/e2e/test_vllm.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+import torch
+from loguru import logger
+
+try:
+    from vllm import LLM, SamplingParams  # type: ignore
+
+    vllm_installed = True
+except ImportError:
+    vllm_installed = False
+    logger.warning("vllm is not installed. This test will be skipped")
+
+
+@pytest.mark.skipif(not vllm_installed, reason="vLLM is not installed, skipping test")
+class TestEagle3vLLM:
+    def setup_method(self):
+        import gc
+
+        gc.collect()
+        if torch.cuda.is_available():
+            torch.cuda.empty_cache()
+
+        self.prompts = [
+            "The capital of France is",
```

Comment:
Utilize more deterministic prompts that don't rely on specific knowledge that may change across models. For example, "Output the following sentence: vLLM is the best inference engine for speculative decoding"